fcrepo-search-impl/src/main/resources/sql/postgresql-search-index.sql

```diff
@@ -0,0 +1,24 @@
+-- simple search
+CREATE TABLE IF NOT EXISTS simple_search (
+    id bigserial PRIMARY KEY,
+    fedora_id  varchar(503) NOT NULL UNIQUE,
+    created timestamp NOT NULL,
+    modified timestamp NOT NULL,
+    content_size bigint DEFAULT NULL,
+    mime_type varchar(255) DEFAULT NULL
+);
+
+CREATE TABLE IF NOT EXISTS search_rdf_type (
+    id bigserial PRIMARY KEY,
+    rdf_type_uri varchar(255) NOT NULL UNIQUE
+);
+
+CREATE TABLE IF NOT EXISTS search_resource_rdf_type (
+    resource_id bigint NOT NULL,
+    rdf_type_id bigint NOT NULL,
+    PRIMARY KEY(resource_id, rdf_type_id),
+    FOREIGN KEY (resource_id) REFERENCES simple_search(id) ON DELETE CASCADE,
+    FOREIGN KEY (rdf_type_id) REFERENCES search_rdf_type(id)
```

Comment:
In the other ddl, this line cascades too. Is there a reason it doesn't here?

Reply:
@pwinckles :  Just a question:  when you have ON DELETE CASCADE on a foreign key,  will the referenced record only be deleted when there are no more references in the link table?  Otherwise, I'm wondering if having  ```FOREIGN KEY (rdf_type_id) REFERENCES search_rdf_type(id) ON DELETE CASCADE ``` might cause an error.

Reply:
@dbernstein That's an excellent question. I'm not certain what will happen.

I'm not sure if these tables should be defined with or without the cascade. I was mostly just drawing attention to the discrepancy in the definitions.